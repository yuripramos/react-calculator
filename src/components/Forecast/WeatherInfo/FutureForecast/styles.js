import styled, { css } from "styled-components";
import { rem, media } from "../../../../styles/tools";
import { white } from "../../../../styles/settings";

export const ContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(10)};
  min-height: ${rem(250)};
  width: 100%;
  display: inline-block;
`;
export const InnerContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(5)};
  min-height: ${rem(250)};
  display: inline-flex;
`;

export const Icon = styled.div`
  width: 100%;
  height: ${rem(60)};
`;

export const DaySpecs = styled.div`
  margin: ${rem(5)} ${rem(13)} ${rem(10)} ${rem(13)};
  ${({ tiny }) =>
    tiny &&
    css`
      padding: ${rem(20)};
      padding: ${rem(15)};
      border: solid 1px rgba(255, 255, 255, 0.2);
      border-radius: 8px;
      width: ${rem(65)};
      margin: ${rem(5)} ${rem(8)} ${rem(5)} ${rem(8)};
    `};
`;

export const Temperature = styled.span`
  font-size: 1.5em;
  font-weight: 200;
  line-height: 0.75;
  font-family: "RobotoSlab Thin";
  margin-top: ${rem(20)};
  width: 100%;
  ${({ tiny }) =>
    tiny &&
    css`
      font-size: 1em;
    `};
`;

export const TinySpecs = styled.span`
  color: white;
  font-weight: 100;
  line-height: 1.1;
  letter-spacing: 0.025em;
  margin: 0;
  padding: 0;
  font-family: "OpenSans Thin";
  width: 100%;
  font-size: 1.17em;
  margin: ${rem(10)} 0;
  :nth-child(2n) {
    margin: ${rem(2)} 0;
  }
  ${({ tiny }) =>
    tiny &&
    css`
      font-size: ${rem(10)};
    `};
`;
