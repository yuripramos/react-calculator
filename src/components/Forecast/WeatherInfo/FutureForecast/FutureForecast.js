import React, { Component } from "react";
import { arrayOf, shape, number, string } from "prop-types";
import moment from "moment";

import IconGenerator from "../../../common/Icon";
import { translate } from "../../../../utils/i18n";
import {
  ContentWrapper,
  Temperature,
  TinySpecs,
  Icon,
  DaySpecs,
  InnerContentWrapper
} from "./styles";
import { Container, Row, Column } from "../../../../styles/grid";

class FutureForecast extends Component {
  render() {
    const { forecast } = this.props;

    return (
      <ContentWrapper>
        <Container>
          <Row>
            <Column>
              <InnerContentWrapper>
                {forecast
                  .filter((j, k) => k > 0)
                  .map((e, i) => (
                    <DaySpecs key={i}>
                      <TinySpecs>
                        {moment.unix(e.time).format("dddd")}
                      </TinySpecs>
                      <Icon>
                        <IconGenerator
                          name={e.icon}
                          width={"70px"}
                          height={"70px"}
                        />
                      </Icon>
                      <Temperature>
                        {`${e.temperatureHigh} ${translate("TEMPERATURE")}`}{" "}
                      </Temperature>
                      <Temperature tiny>
                        {`${e.temperatureLow} ${translate("TEMPERATURE")}`}{" "}
                      </Temperature>
                    </DaySpecs>
                  ))}
              </InnerContentWrapper>
            </Column>
          </Row>
        </Container>
      </ContentWrapper>
    );
  }
}

FutureForecast.defaultProps = {};

FutureForecast.propTypes = {
  forecast: arrayOf(
    shape({
      time: number,
      icon: string,
      temperatureHigh: number,
      temperatureLow: number,
    })),
};

export default FutureForecast;
