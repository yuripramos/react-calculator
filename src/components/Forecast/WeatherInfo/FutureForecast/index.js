import FutureForecast from "./FutureForecast";

export default FutureForecast;
