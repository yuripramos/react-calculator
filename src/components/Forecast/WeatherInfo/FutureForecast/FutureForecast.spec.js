import React from "react";
import { shallow } from "enzyme";

import FutureForecast from "./FutureForecast";

let props;

beforeEach(() => {
  props = {
    forecast: [
      {
        temperatureHigh: 35,
        temperatureLow: 18,
        time: 2312321,
        icon: "clear-night"
      },
      {
        temperatureHigh: 25,
        temperatureLow: 8,
        time: 2314441,
        icon: "clear-day"
      },
      {
        temperatureHigh: 25,
        temperatureLow: 8,
        time: 231111141,
        icon: "clear-day"
      }
    ]
  };
});


describe("FutureForecast Component", () => {
  it("should match snapshot with props", () => {
    expect(shallow(<FutureForecast {...props} />)).toMatchSnapshot();
  });
});
