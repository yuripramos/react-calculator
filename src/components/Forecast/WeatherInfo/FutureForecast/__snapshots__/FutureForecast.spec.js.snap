// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FutureForecast Component should match snapshot with props 1`] = `
<styled.div>
  <Container>
    <Row>
      <Column>
        <styled.div>
          <styled.div
            key="0"
          >
            <styled.span>
              Friday
            </styled.span>
            <styled.div>
              <IconGenerator
                height="70px"
                name="clear-day"
                width="70px"
              />
            </styled.div>
            <styled.span>
              25 undefined
               
            </styled.span>
            <styled.span
              tiny={true}
            >
              8 undefined
               
            </styled.span>
          </styled.div>
          <styled.div
            key="1"
          >
            <styled.span>
              Friday
            </styled.span>
            <styled.div>
              <IconGenerator
                height="70px"
                name="clear-day"
                width="70px"
              />
            </styled.div>
            <styled.span>
              25 undefined
               
            </styled.span>
            <styled.span
              tiny={true}
            >
              8 undefined
               
            </styled.span>
          </styled.div>
        </styled.div>
      </Column>
    </Row>
  </Container>
</styled.div>
`;
